Sources/Sentry/SentryNetworkTracker.m

```diff
@@ -265,6 +288,114 @@ - (void)urlSessionTask:(NSURLSessionTask *)sessionTask setState:(NSURLSessionTas
     SENTRY_LOG_DEBUG(@"SentryNetworkTracker finished HTTP span for sessionTask");
 }
 
+- (void)captureEvent:(NSURLSessionTask *)sessionTask
+{
+    NSHTTPURLResponse *myResponse = (NSHTTPURLResponse *)sessionTask.response;
+    NSNumber *responseStatusCode = [NSNumber numberWithLongLong:myResponse.statusCode];
+
+    if (!self.isCaptureFailedRequests || ![self containsStatusCode:myResponse.statusCode]) {
+        return;
+    }
+
+    if (![self isTargetMatch:sessionTask.currentRequest.URL
+                 withTargets:SentrySDK.options.failedRequestTargets]) {
+        return;
+    }
+
+    NSString *message = [NSString
+        stringWithFormat:@"HTTP Client Error with status code: %ld", (long)myResponse.statusCode];
+
+    SentryEvent *event = [[SentryEvent alloc] initWithLevel:kSentryLevelError];
+
+    SentryThreadInspector *threadInspector = SentrySDK.currentHub.getClient.threadInspector;
```

Comment:
`m`: I would prefer if we passed this in as a dependency.

Reply:
I copied from `SentryANRTrackingIntegration`, which also does not do it, was there a reason?

Reply:
No, not really. I'm a fan of dependency injection to clearly state the required dependencies of a class to work correctly and, furthermore, to improve testability.

Reply:
This will be done along with https://github.com/getsentry/sentry-cocoa/pull/2308#discussion_r1003421632
So the `SentryNetworkTracker` class takes everything via DI.